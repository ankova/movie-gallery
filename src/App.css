@import url("https://fonts.googleapis.com/css?family=Kaushan+Script|Open+Sans&display=swap");
@import url("https://fonts.googleapis.com/css?family=Kaushan+Script|Open+Sans|Saira+Condensed&display=swap");
.App {
  text-align: center; }

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px; }

.App-header {
  font-family: 'Open Sans', sans-serif;
  font-family: 'Kaushan Script', cursive;
  height: 70px;
  padding: 20px;
  color: wheat; }

.App-intro {
  font-size: large; }

.raiting {
  margin-top: 2em;
  display: inline-block;
  width: min-content; }

.raiting input[type="range" i] {
  cursor: grab;
  vertical-align: bottom;
  margin-top: 1em; }

.movie-genres {
  width: 80%;
  margin: auto;
  margin-top: 1em; }

.movie-genres label {
  display: inline-flex;
  margin: 0.2em 0.5em; }

.movie-list {
  display: grid;
  grid-template-columns: auto;
  font-family: 'Open Sans', sans-serif;
  font-family: 'Kaushan Script', cursive;
  font-family: 'Saira Condensed', sans-serif; }

.movie-list ul {
  padding: 0; }

.movie-list ul li {
  box-sizing: border-box;
  max-width: 220px;
  display: inline-block;
  margin: 1em; }

.movie-list ul li img {
  width: 220px; }

.movie-list genres {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: stretch;
  align-content: space-between;
  background: black;
  height: 3em; }

.movie-list genres h6 {
  flex-flow: row;
  flex: auto; }

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg); }
  to {
    transform: rotate(360deg); } }
