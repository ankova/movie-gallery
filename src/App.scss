@import url('https://fonts.googleapis.com/css?family=Kaushan+Script|Open+Sans&display=swap');
@import url('https://fonts.googleapis.com/css?family=Kaushan+Script|Open+Sans|Saira+Condensed&display=swap');

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  font-family: 'Open Sans', sans-serif;
  font-family: 'Kaushan Script', cursive;
  height: 70px;
  padding: 20px;
  color: wheat;
}

.App-intro {
  font-size: large;
}

.raiting {
  margin-top: 2em;
  display: inline-block;
  width: min-content;

  input[type="range" i] {
    cursor: grab;
    vertical-align: bottom;
    margin-top: 1em;
  }
}
.movie-genres {
  width: 80%;
  margin: auto;
  margin-top: 1em;

  label {
    display: inline-flex;
    margin: 0.2em 0.5em;
  }
}

.movie-list {
  display: grid;
  grid-template-columns: auto;
  font-family: 'Open Sans', sans-serif;
  font-family: 'Kaushan Script', cursive;
  font-family: 'Saira Condensed', sans-serif;

  ul {
    padding: 0;

    li {
      box-sizing: border-box;
      max-width: 220px;
      display: inline-block;
      margin: 1em;
  
      img {
        width: 220px;
      }
    }
  }

  genres {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: stretch;
    align-content: space-between;
    background: black;
    height: 3em;

    h6 {
      flex-flow: row;
      flex: auto;
    }
  }
}
@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
