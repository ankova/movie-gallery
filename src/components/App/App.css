/*TO DO
//Extract all fonts and colours in separate file as variables
*/
@import url("https://fonts.googleapis.com/css?family=Kaushan+Script|Open+Sans&display=swap");
@import url("https://fonts.googleapis.com/css?family=Kaushan+Script|Open+Sans|Saira+Condensed&display=swap");
.App {
  text-align: center; }

.App-header {
  font-family: 'Open Sans', sans-serif;
  font-family: 'Kaushan Script', cursive;
  height: 70px;
  padding: 20px;
  color: wheat; }

.movie-genres {
  width: 80%;
  margin: auto;
  margin-top: 1em; }
  .movie-genres label {
    display: inline-flex;
    margin: 0.2em 0.5em; }
