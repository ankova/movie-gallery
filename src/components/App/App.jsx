import React, { useState, useEffect } from 'react';
import axios from 'axios';

import './App.css';
import MovieList from '../MovieList';
import Checkbox from '../Checkbox';
import Rating from '../Rating';

const App = () => {
  const [movies, setMovies] = useState([]); 
  const [genres, setGenres] = useState([]);
  const [selectedGenres, setSelectedGenres] = useState([]);
  const [filteredMovies, setFilteredMovies] = useState([]);
  const [minRating, setMinRating] = useState([]);

  useEffect(() => {
    //Fetch all genres and set to state
    axios.get('https://api.themoviedb.org/3/genre/movie/list?api_key=89c84ad4507eac89b10fd0706f10f092&language=en-US?')
    .then(response => {
      if(response.status === 200) {
      return response.data;
    }})
    .then(response => {
      const genres = response.genres;
      setGenres(genres)

      //Fetch all now playing movies
      axios.get('https://api.themoviedb.org/3/movie/now_playing?api_key=89c84ad4507eac89b10fd0706f10f092&language=en-US&page=1')
      .then(response => {
        if(response.status === 200) {
        return response.data;
      }})
      .then(response => {
            let results = response.results.map(movie =>
                ({ 
                  ...movie,  
                  poster_path: `https://image.tmdb.org/t/p/w500${movie.poster_path}`, //adds full path to poster image
                  //filters all genres by id and adds them to the movie object as a new property
                  genres: genres
                    .filter(genre => movie.genre_ids.indexOf(genre.id) !== -1 )
                    .map(genre => genre.name)
                })
              )
              
            return (results) //returns the modified object
          })
          
      .then(results => setMovies(results)) // adds the modified object to the state
      .catch(error => {console.log(error.message)}) // can be modified to point to an error page if such

    })
    .catch(error => {console.log(error.message)})

  }, []);

  useEffect(() => {
    // sets filtered by genre movies to the state
    setFilteredMovies(
      movies.filter(m => {
        for(let i=0; i<selectedGenres.length; i++){
          const genre = selectedGenres[i];
          if(m.genre_ids.indexOf(genre.id) === -1) {
            return false;
          }
        }
        return true;
      })
      .filter(m => m.vote_average >= minRating) //filter by rating
      .sort((a, b) => b.popularity - a.popularity) // sorts by popularity
    );
  }, [movies, selectedGenres, minRating]); // initial state

  //Filters by genre on checkbox change
  const handleSelectGenre = (event, genre) => { 
    if(event.target.checked) {
      setSelectedGenres([...selectedGenres, genre]);
    } else {
      setSelectedGenres(selectedGenres.filter(g => g.id !== genre.id))
    }
  }

  //Filters by rating
  const handleRatingChange = (rating) => {
    setMinRating(rating);
  }

    return ( 
      <div className="App">
        <div className="App-header">
          <h2>Playing Movies</h2>
        </div>

        <div className="movie-genres">
          {genres.map(genre => 
            <Checkbox key={genre.id} name={genre.name} onChange={e => handleSelectGenre(e, genre)} />)}
        </div>
         
        <Rating onChange={handleRatingChange} />

        <MovieList movies={filteredMovies} />
        
      </div>
    );
  }

export default App;
