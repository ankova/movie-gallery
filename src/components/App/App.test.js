import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import axios from 'axios';

import App from './App';

describe('App component', () => {
  let component;
  const movies = [{
    name: "Joni Baez",
    genres: ["action", 'triller', 'horror']
  }];

  beforeEach(() => component = mount(<App />));

  it('renders component without crashing', () => {
    const tree = renderer.create(<App />).toJSON;
    expect(tree).toMatchSnapshot();
  });

  it('tests', () => {
    axios.get = jest.fn();
    const setMovies = jest.fn().mock;
    const test = axios.get.mockImplementation(() => Promise.resolve(movies));
    // const setMovies = jest.fn().mockImplementation(test);
    // expect(setMovies).toBeCalled();

    console.log(test())
    console.log(setMovies)
    console.log(component.html())

  });

});


