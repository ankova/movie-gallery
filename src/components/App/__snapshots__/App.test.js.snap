exports[`App component renders component without crashing 1`] = `[Function]`;
